bmorph/util/mizuroute_utils.py

```diff
@@ -0,0 +1,217 @@
+    routed['cdf_blend_factor'].values = (routed['distance_to_up_gauge'] 
+                                         / (routed['distance_to_up_gauge'] + routed['distance_to_down_gauge'])).values
+    routed['cdf_blend_factor'] = routed['cdf_blend_factor'].where(~np.isnan(routed['cdf_blend_factor']), other=0.0)
+    
+    for site, seg in site_2_seg.items():
+        routed['downstream_ref_seg'].loc[{'seg':seg}] = seg
+        routed['upstream_ref_seg'].loc[{'seg':seg}] = seg
+    
+    for seg in routed['seg'].values:
+        if seg in seg_2_site.keys() or routed.sel(seg=seg)['down_seg'].values[()] not in routed['seg'].values:
+            continue
+        current_seg = seg
+        while routed.sel(seg=current_seg)['down_seg'].values[()] not in seg_2_site.keys():
+            current_seg = routed.sel(seg=current_seg)['down_seg'].values[()]
+        downstream_gauge_seg = routed.sel(seg=current_seg)['down_seg'].values[()]
+        routed['downstream_ref_seg'].loc[{'seg': seg}] = downstream_gauge_seg
+        
+    for seg in routed['seg'].values:
+        if seg in seg_2_site.keys() or routed.sel(seg=seg)['up_seg'].values[()] not in routed['seg'].values:
+            continue
+        current_seg = seg
+        while routed.sel(seg=current_seg)['up_seg'].values[()] not in seg_2_site.keys():
+            current_seg = routed.sel(seg=current_seg)['up_seg'].values[()]
+            if current_seg is None:
+                break
+        if current_seg is None:
+            routed['upstream_ref_seg'].loc[{'seg':seg}] = routed['downstream_ref_seg'].loc[{'seg': seg}]
+        else:
+            upstream_gauge_seg = routed.sel(seg=current_seg)['up_seg'].values[()]
+            routed['upstream_ref_seg'].loc[{'seg': seg}] = upstream_gauge_seg
+        
+    routed['downstream_ref_seg'] = (routed['downstream_ref_seg'].where(
+        ~np.isnan(routed['downstream_ref_seg']), other=routed['upstream_ref_seg']))
+    routed['upstream_ref_seg'] = (routed['upstream_ref_seg'].where(
+        ~np.isnan(routed['upstream_ref_seg']), other=routed['downstream_ref_seg']))
+    
+    return routed
+    
+def annotate_variable(routed: xr.Dataset, condition: xr.Dataset, var_label: str, condition_var: str, 
+                      gauge_segs: list):
+    """
+    splits the variable into its upstream and downstream components to be used in blendmorph
+    ----
+    routed: xr.Dataset
+        the dataset that will be modified and returned having been prepared by annotate_reaches
+        with the dimension 'seg'
+    condition: None
+        contains the variable to be split into upstream and downstream components and can be 
+        the same as routed, (must also contain the dimension 'seg')
+    var_label: str
+        suffix of the upstream and downstream parts of the variable
+    condition_var: str
+        variable name of the variable to be split in upstream and downstream parts
+    gauge_segs: list
```

Comment:
It might be worth making this optional, and if not provided then it just uses the segs in the `routed` dataset.

Reply:
made it optional where it just pulls the `gauge_segs` from `is_gauge` if nothing else is given